File: Python/program/Tabel.py
# Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
# soal Latihan Praktikum 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin


class Tabel:
    __baris = ""
    __kolom = ""

    def __init__(self, baris=0, kolom=0):
        self.__baris = baris
        self.__kolom = kolom

    def get_baris(self):
        return self.__baris

    def set_baris(self, baris):
        self.__baris = baris

    def get_kolom(self):
        return self.__kolom

    def set_kolom(self, kolom):
        self.__kolom = kolom

    def buat_garis(self, max_len):
        for i in range(self.get_kolom()):
            print("+", end="")
            for j in range(max_len[i] + 2):
                print("-", end="")
        print("+")

    def find_max_len(self, isi):
        max_len = [0] * self.get_kolom()
        for i in range(self.get_kolom()):
            for j in range(self.get_baris()):
                if len(isi[j][i]) > max_len[i]:
                    max_len[i] = len(isi[j][i])
        return max_len

    def buat_tabel(self, isi):
        max_len = self.find_max_len(isi)

        self.buat_garis(max_len)

        for j in range(self.get_kolom()):
            print("| " + isi[0][j], end="")
            for k in range(max_len[j] - len(isi[0][j])):
                print(" ", end="")
            print(" ", end="")
        print("|")

        self.buat_garis(max_len)

        for i in range(1, self.get_baris()):
            for j in range(self.get_kolom()):
                print("| " + isi[i][j], end="")
                for k in range(max_len[j] - len(isi[i][j])):
                    print(" ", end="")
                print(" ", end="")
            print("|")

        self.buat_garis(max_len)

File: CPP/program/Tabel.cpp
/*Saya Boy Aditya Rohmaulana NIM 2203488 mengerjakan
soal Latihan Praktikum 4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#include <iostream>
#include <string>

using namespace std;

class Tabel {
  private:
    int baris;
    int kolom;

  public:
    //   Constructor
    Tabel() {
    }

    Tabel(int baris, int kolom) {
        this->baris = baris;
        this->kolom = kolom;
    }

    //  Getter and Setter
    int getBaris() {
        return this->baris;
    }

    void setBaris(int baris) {
        this->baris = baris;
    }

    int getKolom() {
        return this->kolom;
    }

    void setKolom(int kolom) {
        this->kolom = kolom;
    }

    //  Method
    void buatGaris(int maxLen[]) {
        for (int i = 0; i < getKolom(); i++) {
            cout << "+";
            for (int j = 0; j < maxLen[i] + 2; j++) {
                cout << "-";
            }
        }
        cout << "+" << endl;
    }

    void findMaxLen(string isi[][9], int maxLen[]) {
        for (int i = 0; i < getKolom(); i++) {
            maxLen[i] = 0;
            for (int j = 0; j < getBaris(); j++) {
                if (isi[j][i].length() > maxLen[i]) {
                    maxLen[i] = isi[j][i].length();
                }
            }
        }
    }

    void buatTabel(string isi[][9]) {
        int maxLen[9];

        findMaxLen(isi, maxLen);

        buatGaris(maxLen);

        for (int j = 0; j < getKolom(); j++) {
            cout << "| " << isi[0][j];
            for (int k = 0; k < maxLen[j] - isi[0][j].length(); k++) {
                cout << " ";
            }
            cout << " ";
        }
        cout << "|" << endl;

        buatGaris(maxLen);

        for (int i = 1; i < getBaris(); i++) {
            for (int j = 0; j < getKolom(); j++) {
                cout << "| " << isi[i][j];
                for (int k = 0; k < maxLen[j] - isi[i][j].length(); k++) {
                    cout << " ";
                }
                cout << " ";
            }
            cout << "|" << endl;
        }

        buatGaris(maxLen);
    }

    // Destructor
    ~Tabel() {
    }
};